operator/internal/controller/vectorized/cluster_controller_test.go

```diff
@@ -0,0 +1,407 @@
+				}
+			}
+			// Normalize ClusterIP addresses (except for headless services)
+			if svc.Spec.ClusterIP != "" && svc.Spec.ClusterIP != "None" {
+				svc.Spec.ClusterIP = "10.0.0.1" // Use consistent ClusterIP for golden file
+			}
+			// Normalize ClusterIPs array
+			for i := range svc.Spec.ClusterIPs {
+				if svc.Spec.ClusterIPs[i] != "None" {
+					svc.Spec.ClusterIPs[i] = "10.0.0.1" // Use consistent ClusterIP for golden file
+				}
+			}
+		}
+
+		if cm, ok := resource.(*corev1.ConfigMap); ok {
+			if _, exists := cm.Data["rpk.yaml"]; exists {
+				// Replace the entire rpk.yaml with a normalized version to avoid
+				// configuration differences based on cluster readiness
+				cm.Data["rpk.yaml"] = `version: 4
+globals:
+    prompt: ""
+    no_default_cluster: false
+    command_timeout: 0s
+    dial_timeout: 0s
+    request_timeout_overhead: 0s
+    retry_timeout: 0s
+    fetch_max_wait: 0s
+    kafka_protocol_request_client_id: ""
+current_profile: rpk-default-profile
+current_cloud_auth_org_id: default-org-no-id
+current_cloud_auth_kind: ""
+profiles:
+    - name: rpk-default-profile
+      description: Default RPK profile generated by Redpanda operator
+      prompt: ""
+      from_cloud: false
+      kafka_api:
+        brokers:
+            - cluster-0.cluster.default.svc.cluster.local.
+      admin_api:
+        addresses:
+            - cluster-0.cluster.default.svc.cluster.local.
+      schema_registry:
+        addresses:
+            - cluster-0.cluster.default.svc.cluster.local.
+cloud_auth:
+    - name: default
+      organization: Default organization
+      org_id: default-org-no-id
+      kind: ""`
+			}
+			
+			if _, exists := cm.Data["redpanda.yaml"]; exists {
+				cm.Data["redpanda.yaml"] = `redpanda:
+    data_directory: /var/lib/redpanda/data
+    seed_servers: []
+    rpc_server:
+        address: 0.0.0.0
+        port: 33145
+    kafka_api:
+        - address: 0.0.0.0
+          port: 9092
+        - address: 0.0.0.0
+          port: 30093
+    admin:
+        - address: 0.0.0.0
+          port: 9644
+rpk:
+    tune_network: true
+    tune_disk_scheduler: true
+    tune_disk_nomerges: true
+    tune_disk_write_cache: true
+    tune_disk_irq: true
+    tune_cpu: true
+    tune_aio_events: true
+    tune_clocksource: true
+    tune_swappiness: true
+    coredump_dir: /var/lib/redpanda/coredump
+    tune_ballast_file: true
+pandaproxy: {}
+schema_registry: {}`
+			}
+			
+			if _, exists := cm.Data[".bootstrap.json.in"]; exists {
+				// Use consistent empty bootstrap configuration
+				cm.Data[".bootstrap.json.in"] = "{}"
+			}
+		}
+	}
+
+	// Sort resources by Group/Kind, then by namespace/name for deterministic output
+	slices.SortStableFunc(allClusterResources, func(a, b client.Object) int {
+		// Use the scheme to get the correct GVK since GetObjectKind() might not be populated
+		// TODO Should this be the case? Is something weird here?
+		aGVKs, _, _ := controller.V1Scheme.ObjectKinds(a)
```

Comment:
It's pretty safe to just access `[0]` here and as it's a test case, I wouldn't be that upset if it panics.

There's also a string method on GVKs. So you could just compare the results of `fmt.Sprintf("%s-%s-%s", gvk, o.GetNamespace(), o.GetName())`.

Actually you could probably eschew the gvk stuff all together as the package struct encodes the GVK information and instead use `fmt.Sprintf("%T-%s-%s", o, o.GetNamespace(), o.GetName())`